cli/cmd/version.go

```diff
@@ -67,8 +88,123 @@ func IsHappyOutdated(cmd *cobra.Command) (bool, *util.Release, *util.Release, er
+		//return false, cliVersion, latestAvailableVersion, err
+		return false, cliVersion, cliVersion, nil // Lie.
 	}
 
 	return !cliVersion.Equal(latestAvailableVersion), cliVersion, latestAvailableVersion, nil
 }
+
+func WarnIfHappyOutdated(cmd *cobra.Command) {
+
+	outdated, cliVersion, latestAvailableVersion, err := IsHappyOutdated(cmd)
+
+	if err != nil {
+		log.Errorf("Error checking for latest available version number: %v", err)
+		return
+	}
+
+	if outdated {
+		log.Warnf("This copy of Happy CLI is not the latest available. CLI version: %s  Latest available version: %s\n", cliVersion.Version, latestAvailableVersion.Version)
+		log.Warn("To update on Mac, run:  brew upgrade happy")
+	}
+
+}
+
+func CreateHappyVersionLockfileHandler(cmd *cobra.Command) error {
+	happyConfig, err := config.GetHappyConfigForCmd(cmd)
+	if err != nil {
+		return err
+	}
+
+	currentVersion := util.GetVersion()
+	projectRoot := happyConfig.GetProjectRoot()
+
+	requestedVersion, _ := cmd.Flags().GetString("version")
+
+	if requestedVersion == "" {
+		requestedVersion = currentVersion.Version
+	}
+
+	path, version, err := createHappyVersionLockFile(projectRoot, requestedVersion)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Created %s locking Happy to version %s\n", path, version)
+
+	return nil
+
+}
+
+func createHappyVersionLockFile(projectRoot string, requestedVersion string) (string, string, error) {
+
+	versionFile, err := config.NewHappyVersionLockFile(projectRoot, requestedVersion)
+	if err != nil {
+		return "", "", err
+	}
+
+	err = versionFile.Save()
+	if err != nil {
+		return "", "", err
+	}
+
+	return versionFile.Path, versionFile.HappyVersion, nil
+}
+
+func VerifyHappyIsLockedVersion(cmd *cobra.Command) (bool, string, string, error) {
+	happyConfig, err := config.GetHappyConfigForCmd(cmd)
+	if err != nil {
+		return false, "", "", err
+	}
+
+	projectRoot := happyConfig.GetProjectRoot()
+
+	/*
+		For backward compatibility reasons, if the .happy/version.lock file does not exist,
+		we will simply return true. We are essentially saying that an unlocked version is the same
+		as when an version is locked and matched.
+	*/
+
+	if !config.DoesHappyVersionLockFileExist(projectRoot) {
+		return true, "", "", nil
+	}
+
+	happyVersionLock, err := config.LoadHappyVersionLockFile(projectRoot)
+	if err != nil {
+		return false, "", "", err
+	}
+
+	if util.GetVersion().Version != happyVersionLock.HappyVersion {
+		return false, util.GetVersion().Version, happyVersionLock.HappyVersion, nil
+	}
+
+	return true, util.GetVersion().Version, happyVersionLock.HappyVersion, nil
+}
+
+func CheckLockedHappyVersion(cmd *cobra.Command) error {
+
+	excludeLockedVersionCheckCmds := map[string]interface{}{
+		"version":           nil,
+		"set-lock":          nil,
+		"available-version": nil,
+	}
+
+	log.Debugf("Current command: %s\n", cmd.CalledAs())
+	if _, present := excludeLockedVersionCheckCmds[cmd.CalledAs()]; !present {
```

Comment:
```suggestion
	if _, present := excludeLockedVersionCheckCmds[cmd.CalledAs()]; present {
```
refactor to return fast if not present and have the main body of logic handle the case when the cmd is in the map

Reply:
ahh here is the style guide I use : https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

Reply:
Done